flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferWritingResultPartition.java

```diff
@@ -330,9 +330,12 @@ private BufferBuilder appendBroadcastDataForRecordContinuation(
     private void createBroadcastBufferConsumers(BufferBuilder buffer, int partialRecordBytes)
             throws IOException {
         try (final BufferConsumer consumer = buffer.createBufferConsumerFromBeginning()) {
+            int desirableBufferSize = Integer.MAX_VALUE;
             for (ResultSubpartition subpartition : subpartitions) {
-                subpartition.add(consumer.copy(), partialRecordBytes);
+                int subPartitionBufferSize = subpartition.add(consumer.copy(), partialRecordBytes);
```

Comment:
If `subpartition#add` fails it returns the negative value(`-1`), so we should take into account this case and for example ignore such value.

Reply:
Thank you very much for review. I add special treatment for negative value

Reply:
Thank you very much for review. I add special treatment for negative value

Reply:
I believe more correctly is to ignore the value for calculation of desirable buffer size rather than ignore the total value. I mean:
```
if(subPartitionBufferSize > 0) {
   desirableBufferSize = Math.min(desirableBufferSize, subPartitionBufferSize);
}
```
I think that it is better because if one of the subpartitions fails we are still able to send it to another ones. As I understand eventually they all will be closed if at least one was closed so maybe it is not so important but anyway it is better to follow the current semantic which doesn't forbid sending data to subpartitions even if one of them is closed.